uktrade/data-hub-frontend
# Feature/task deletion

## Description of change

Combination of previously reviewed sub tickets into feature branch.

Replace completing of tickets using archived with status is complete.
Add delete functionality using archived.

## Description of change

Update the FE to use the new status field and allow for Deletion of Tasks using archived.

On the my tasks dashboard deleted tasks should be hidden.

On the task details page, show DELETED status on the task - this is so users can click through from ‘task deleted by other user’ notification.

Add Delete button on Task view page using.

## Test instructions
Run this locally with the [feature/task-deletion](https://github.com/uktrade/data-hub-api/pull/5198) branch on the API.

- Whilst testing verify changes below are reflected on the Task dashboards (please note this uses Open search and there may be a delay in changes being reflected).
- Whilst testing verify that reminders are send for each of the changes via gov notify (changes are scheduled using long/short queues). 

1. Create a Task from the Tasks tab on the Data Hub homepage.
2. Edit the task
3. From the edit page mark the task as complete. [The task should be marked complete]
4. From the edit page mark the task as deleted. [The task should be deleted]
5. Using the back button of your browser go back to the deleted task, it should be marked Deleted.
6. The deleted task should not be included in the Tasks tab table.

## Screenshots

### Before
No delete button visible.

![image](https://github.com/uktrade/data-hub-frontend/assets/699259/0ebb8d90-1b3c-446f-a720-aae85f22fb07)

### After
A delete button should be displayed on the Edit Task page.

![image](https://github.com/uktrade/data-hub-frontend/assets/699259/ad36f758-0a0b-496a-92a5-97138483ece6)

### Dashboard
No visual changes in the my tasks dashboard

![image](https://github.com/uktrade/data-hub-frontend/assets/699259/cff797d8-0a21-4988-b7f2-ff83ece2ae18)
 

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
